typescript/forms.d.ts

```diff
@@ -1132,7 +1132,54 @@ export type BuiltInFormElementLookup = NewBuiltInFormElementLookup & {
 }
 
 export type FormElementLookupEnvironmentUrl = FormElementEnvironmentUrl
-export type NewFormElementLookup = {
+
+export type FormElementLookupTableRowBase = {
+  // value entered into the element used to trigger the lookup
+  inputValue: string
+}
+
+export type FormElementTableRowText = FormElementLookupTableRowBase & {
+  // the value to populate the configured element with
+  preFillValue: string
+}
+
+export type FormElementTableRowNumber = FormElementLookupTableRowBase & {
+  // The value to prefil the correpsonding column elementName with
+  preFillValue: number
+}
+
+export type FormElementLookupTableColumnBase = {
+  // the name of the form element to prefill
+  elementName: string
+}
+
+export type FormElementLookupTableColumnText = {
+  type: 'TEXT'
+  /// array of rows for each value matching agaisnt the "inputValue" from the element
+  rows: FormElementTableRowText[]
+}
+
+export type FormElementLookupTableColumnNumber = {
+  type: 'NUMBER'
+  /// array of rows for each value matching agaisnt the "inputValue" from the element
+  rows: FormElementTableRowNumber[]
+}
+
+export type FormElementLookupTableEnvironment = {
+  formsAppEnvironmentId: number
+  // array of elements with prefil values for each corresponding input value
+  columns: (
+    | FormElementLookupTableColumnText
+    | FormElementLookupTableColumnNumber
+  )[]
```

Comment:
```suggestion
  columns: Array(
    | FormElementLookupTableColumnText
    | FormElementLookupTableColumnNumber
  >
```

This is just personal preference

Reply:
Agree it is easier to read